src/main/java/seedu/address/commons/util/StringUtil.java

```diff
@@ -60,7 +60,20 @@ public static boolean isNonZeroUnsignedInteger(String s) {
 
         try {
             int value = Integer.parseInt(s);
-            return value > 0 && !s.startsWith("+"); // "+1" is successfully parsed by Integer#parseInt(String)
+            // "+1" is successfully parsed by Integer#parseInt(String)
+            return value > 0 && !s.startsWith("+") && !s.contains("-");
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+    }
+
+    public static boolean isNonNegativeInteger(String s) {
+        requireNonNull(s);
+
+        try {
+            int value = Integer.parseInt(s);
+            // "+1" is successfully parsed by Integer#parseInt(String)
+            return value >= 0 && !s.startsWith("+") && !s.contains("-");
```

Comment:
Also apart from adding the appropriate tests for hourlySalary and the other attributes using this method, maybe you could add tests for this method by itself? I think it would also be another chance to show that you know how to make test cases.